### PR TITLE
Show current location on target points screen

### PR DESCRIPTION
**Pull request by kevquincke**
#### Trello board reference:

* https://trello.com/c/uU5lDNKc/3-3-as-a-user-i-should-be-able-to-see-my-current-location-on-the-map

---

#### Description:

* As a user I should be able to see my current location on the map.
* Buttons on the nav bar will not be clickeable for now.
* Create new target will not be clickeable for now.

### Screenshot:
show current location

### CODE REVIEW

File: src/components/Common/CustomButton/index.js [hunk 1]
```diff
@@ -31,10 +35,12 @@ const CustomButton = ({ title,
 };
 
 CustomButton.propTypes = {
-  title: string.isRequired,
+  title: string,
   onPress: func.isRequired,
   style: any,
   fontStyle: any,
+  image: any,
```
Comment (bernabe9) [inline src/components/Common/CustomButton/index.js]: why `any`?

File: src/components/Common/CustomMap/index.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+  componentWillMount() {
+    this.getCurrentPosition(this.onRegionChange);
+  }
+
+  componentDidMount() {
+    this.getCurrentPosition((coords) => {
+      const location = {
+        latitude: coords.latitude,
+        longitude: coords.longitude,
+        latitudeDelta: 0.0035,
```
Comment (bernabe9) [inline src/components/Common/CustomMap/index.js]: create a new constant for delta's value

File: src/components/Common/CustomMap/index.js [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+  render() {
+    const { style } = this.props;
+    const { location, error } = this.state;
+    const { container, containerError, map } = styles;
+    const containerStyles = StyleSheet.flatten([!error ? container : containerError, style]);
+
+    return (
+      <View style={containerStyles}>
+        {!error && <MapView
```
Comment (bernabe9) [inline src/components/Common/CustomMap/index.js]: the correct alignment should be:

```javascript
{!error &&
  <MapView
    ...
  >
  </MapView>
}
```

File: src/components/Common/CustomMap/index.js [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+          showsPointsOfInterest={false}
+          followsUserLocation
+          onRegionChange={() => this.getCurrentPosition(this.onRegionChange)}
+          style={map}
+        >
+          <Marker coordinate={location}>
+            <Image source={marker} />
+          </Marker>
+        </MapView>}
+        {error && <Text style={styles.error}>
```
Comment (bernabe9) [inline src/components/Common/CustomMap/index.js]: same

**Comment by aguscha333** [comment]
Nice work! I don't have anything to add to @bernabe9 's comments. Will approve once those are fixed.

**Comment by kevquincke** [comment]
@bernabe9 @aguscha333 all issues have been resolved. Image now is using node type, constants have been created and code indented the right way. Thank you for the review! 😄

File: src/containers/TargetPointsScreen/index.js [hunk 5]
```diff
@@ -16,25 +38,44 @@ class TargetPointsScreen extends Component {
...
+        />
       </View>
     );
   }
 }
 
 TargetPointsScreen.propTypes = {
-  signOut: func.isRequired
+  signOut: func.isRequired,
+  navigator: object
```
Comment (missx) [inline src/containers/TargetPointsScreen/index.js]: Navigator should be required, right?

Comment (kevquincke) [reply src/containers/TargetPointsScreen/index.js]: Indeed, thank you! I've set it to required

**Pull request merged**